src/lib.rs

```diff
@@ -148,30 +166,9 @@ impl Kaboodle {
             .await
             .insert(self_addr, PeerState::Known(Instant::now()));
 
-        // Set up our broadcast communications socket
-        // For listening to broadcasts, we need to bind our socket to 0.0.0.0:<port>, but for
-        // sending broadcasts, we need to send to 255.255.255.255:<port>.
-        let broadcast_inbound_addr = SocketAddr::V4(SocketAddrV4::new(
-            Ipv4Addr::new(0, 0, 0, 0),
-            self.broadcast_port,
-        ));
-        let broadcast_outbound_addr = SocketAddr::V4(SocketAddrV4::new(
-            Ipv4Addr::new(255, 255, 255, 255),
-            self.broadcast_port,
-        ));
-        let broadcast_sock = {
-            let broadcast_sock =
-                Socket::new(Domain::IPV4, Type::DGRAM, Some(Protocol::UDP)).unwrap();
-            broadcast_sock.set_broadcast(true).unwrap();
-            broadcast_sock.set_nonblocking(true).unwrap();
-            broadcast_sock.set_reuse_address(true).unwrap();
-            broadcast_sock.set_reuse_port(true).unwrap();
-            broadcast_sock
-                .bind(&SockAddr::from(broadcast_inbound_addr))
-                .expect("Failed to bind for broadcast");
-            let broadcast_sock: std::net::UdpSocket = broadcast_sock.into();
-            UdpSocket::from_std(broadcast_sock).unwrap()
-        };
+        // Set up our broadcast communications sockets
+        let (broadcast_in_sock, broadcast_out_sock, broadcast_addr) =
```

Comment:
I want to call out the fact that doing multicasting in IPv6 requires two separate sockets whereas IPv4 could do everything with one, which is something that took me ages to figure out.

For the sake of simplicity, IPv4 is now using two as well.

Reply:
Yeah. :/ I went through a little emotional wander and considered abstracting out a `Broadcaster` instance that you'd hand an interface, IP address, and port number to, and it would expose send and receive methods. I might still do that someday, but it didn't seem worth pushing on yet.